File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.py
from itertools import permutations

# 소수확인 함수
def is_snum(x):
    if x < 2 :
        return False
    
    for i in range(2, x) :
        if x % i == 0 :
            return False
            
    return True

def solution(numbers):
    answer = 0
    nums = []
    
    # 각각 조합하기
    for i in range(1, len(numbers)+1) :
        nums.append(list(set(map(''.join, permutations(numbers, i)))))
    per = list(set(map(int, set(sum(nums, [])))))
    
    for p in per:
        if is_snum(p) == True:
            answer += 1

    return answer

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.js
function solution(numbers) {
    var answer = 0;
    var set = new Set();
    makeNumbers(set , '' , numbers.split(''));
    return set.size;
}

function issosu(num) {
    if( num < 2) return false;
    for (var i =2; i <= num / 2 ; i++) {
        if( num % i === 0) return false;
    }
    return true;
}

function makeNumbers(set , cur, nums) {
    if( nums.length === 0 ) return;
    var clone = nums.slice().reverse();
    nums.forEach(function(i) {
        var su = clone.pop();
        var num = Number(cur+su);
        if ( issosu(num)) {
            set.add(num);
        }

        makeNumbers(set, cur+su , clone);
        clone.unshift(su);
    }) 
}